bgpd/bgp_network.c

```diff
@@ -909,7 +909,7 @@ void bgp_close(void)
 	for (ALL_LIST_ELEMENTS(bm->listen_sockets, node, next, listener)) {
 		if (listener->bgp)
 			continue;
-		thread_cancel(listener->thread);
+		THREAD_OFF(listener->thread);
```

Comment:
shouldn't `bgp_close_vrf_socket()` be updated to use `THREAD_OFF` as well?

Reply:
Just noticed Quentin is already doing that on #6467 while reviewing his PR. Marking as resolved...